lib/rbac/filterer.rb

```diff
@@ -424,6 +424,10 @@ def scope_to_tenant(scope, user, miq_group)
       user_or_group = user || miq_group
       tenant_id_clause = klass.tenant_id_clause(user_or_group)
 
+      if klass.respond_to?(:tenant_joins_clause)
```

Comment:
@rwsu I am just thinking about.
this is supporting for supporting origin cloud_tenancy_mixin and tenancy mixin at the same time ? 

Should not it be exclusive ? only either  cloud_tenancy_mixin or  only either tenancy_mixin ?
If so I would create in cloud tenant mixin 
method `:scope_by_cloud_tenant?`
and in filterer move out this condition from `scope_to_tenant`
and add elsif branch for in `scope_targets` method 
```ruby 
      if klass.respond_to?(:scope_by_tenant?) && klass.scope_by_tenant?
        scope = scope_to_tenant(scope, user, miq_group)
     elsif klass.try(:scope_by_cloud_tenant?)
         scope =  klass.tenant_joins_clause(scope)
      end
```

What do you think ? I think we should separate this two things in RBAC if it will be work.